app/models/katello/content_view.rb

```diff
@@ -794,12 +794,20 @@ def last_publish_task_success?
       return last_publish_result.present? && last_publish_result == 'success'
     end
 
+    def cv_repo_indexed_after_last_published?
+      repositories.any? { |repo| repo.last_indexed && repo.last_indexed > latest_version_object.created_at }
+    end
+
     def needs_publish?
       #Returns
       # True:
       #     a) When content/repo/filter change audit records exist
       #     b) CV hasn't ever been published
       #     c) CV dependency_solving != latest_version.applied_filters.dependency_solving
+      #     d) If repo was indexed after cv publish. This can happen under 3 cases:
+      #       i) Index runs because last index(before publish) had failed and repo is picked up for index even if pulp publication hasn't changed.
+      #       ii) Complete sync runs or sync adds/removes new content (Already true because new pulp publication/version gets created)
+      #       iii) repo.index_content(force_index: true) is run. (This doesn't necessarily indicate contents changed. Corner case where we play safe and return true)
```

Comment:
```suggestion
      #       iii) repo.index_content is run. (This doesn't necessarily indicate contents changed. Corner case where we play safe and return true)
```

From the code and testing it looks like `force_index: true` isn't needed?

Reply:
Hah..I had that meant for the action Repository::IndexContent which takes that param but then I moved the indexed_at datetime update to index_content method to be more accurate..Will update the text here.